python/lsst/afw/detection/footprintArray.cc

```diff
@@ -21,17 +21,53 @@
  */
 
 #include <pybind11/pybind11.h>
-//#include <pybind11/operators.h>
 //#include <pybind11/stl.h>
 
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+#include "ndarray/converter.h"
+
+#include "lsst/afw/detection/Footprint.h"
+#include "lsst/afw/detection/FootprintArray.cc"     // FootprintArray.h does not define the templates
+
 namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace afw {
+namespace detection {
 
-using namespace lsst::afw::detection;
+namespace {
+    template <typename T>
+    void declareFlattenArray(py::module & mod) {
+        namespace afwGeom = lsst::afw::geom;
+
+        mod.def("flattenArray", (void (*)(Footprint const &,
+                                          ndarray::Array<T const, 2, 0> const &,
+                                          ndarray::Array<T, 1, 0> const &,
+                                          afwGeom::Point2I const &)) &flattenArray<T const, T, 2, 0, 0>,
+                "fp"_a, "src"_a, "dest"_a, "xy0"_a=afwGeom::Point2I());
+        mod.def("flattenArray", (void (*)(Footprint const &,
+                                          ndarray::Array<T const, 3, 0> const &,
+                                          ndarray::Array<T, 2, 0> const &,
+                                          afwGeom::Point2I const &)) &flattenArray<T const, T, 3, 0, 0>,
+                "fp"_a, "src"_a, "dest"_a, "xy0"_a=afwGeom::Point2I());
+    }
+}
 
 PYBIND11_PLUGIN(_footprintArray) {
     py::module mod("_footprintArray", "Python wrapper for afw _footprintArray library");
 
+    if (_import_array() < 0) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        return nullptr;
+    }
+
     /* Module level */
+    declareFlattenArray<std::uint16_t>(mod);
+    declareFlattenArray<int>(mod);
+    declareFlattenArray<float>(mod);
+    declareFlattenArray<double>(mod);
```

Comment:
@kfindeisen has already wrapped this on his ticket. As I said in his review: could one of you please also wrap `expandArray`? It is required by at least two packages downstream. I believe it is wrapped similarly (perhaps identically) to `flattenArray` and if so, I suggest renaming this to `declareAll` and declaring both here.

It appears that many of the remaining changes are also duplicated in DM-8419...